packages/cloud-functions/src/cico/logParser/parseEventLogs.ts

```diff
@@ -0,0 +1,78 @@
+import { readFileSync } from 'fs'
+import { Providers } from '../Providers'
+
+// Inputs:
+const logFilePath = '/Users/tarik/Downloads/moonpay_logs.json' // Your local log file here
+const provider = Providers.Moonpay // The provider which is getting logs parsed
+
+const logs = JSON.parse(readFileSync(logFilePath, 'utf8'))
+
+const project = logs[0].logName.split('/')[1]
+process.env.GCLOUD_PROJECT = project
+// Using an in-line require because process variables need to be defined
+// before calling these functions
+const { parseMoonpayEvent, MOONPAY_BIG_QUERY_EVENT_TABLE } = require('../moonpayWebhook')
+const { parseRampEvent, RAMP_BIG_QUERY_EVENT_TABLE } = require('../rampWebhook')
+const { parseXanpoolEvent, XANPOOL_BIG_QUERY_EVENT_TABLE } = require('../xanpoolwebhook')
+const { parseTransakEvent, TRANSAK_BIG_QUERY_EVENT_TABLE } = require('../transakWebhook')
+
+const selectEventParser = (provider: Providers): Function => {
+  if (provider === Providers.Moonpay) {
+    return parseMoonpayEvent
+  }
+
+  if (provider === Providers.Ramp) {
+    return parseRampEvent
+  }
+
+  if (provider === Providers.Xanpool) {
+    return parseXanpoolEvent
+  }
+  if (provider === Providers.Transak) {
+    return parseTransakEvent
+  }
+
+  return () => ({})
+}
+
+const selectEventTable = (provider: Providers): string => {
```

Comment:
Maybe we can combine this with `selectEventParser` and return both things in a `{ parser, eventTable }` object?

Reply:
Yep good suggestion